### DISCUSSION TITLE
Save changes alert box.

### DISCUSSION THREAD

**Discussion by setanarut** [sbarex/SourceCodeSyntaxHighlight#120] [Ideas]
After putting effort and editing the theme. All changes will be lost if you close the window. just happened to me. I closed the window thinking it was autosaved.

**Comment by setanarut** [comment]
this independent box is a bit odd. It's not like the macOS experience. The notification message does not need to be a popup. I think it should be the save warning like save window in TextEdit. and Command+s should not wait for confirmation.
<img width="295" alt="Ekran Resmi 2021-11-15 18 50 29" src="https://user-images.githubusercontent.com/36481442/141812424-0d9511cd-66d4-4501-8bbd-f9e234cf775b.png">
.

**Comment by sbarex** [comment]
Normally, if there are unsaved settings, a warning is shown when the application is closed. There is a bug that doesn't set dirty status when themes are changed. Will be fixed in the next release.

**Comment by sbarex** [comment]
On the release 2.1.2 I fix the bug. Now the alet box with the save notification will appear only with the debug option.

**Reply by setanarut** [reply]
Updating from the menu doesn't work.

**Reply by setanarut** [reply]
latest 2.1.2 is not 2.1.2.
<img width="548" alt="Ekran Resmi 2021-11-19 20 53 36" src="https://user-images.githubusercontent.com/36481442/142669031-d9603d80-3995-45ef-b777-78ce50d3c102.png">

**Comment by sbarex** [comment]
fixed
